### PR TITLE
feat: Improve new Sell flow bottom navigation

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

This PR solves [ONYX-1072]

### Description

This improves the new Sell flow bottom navigation:

-  Saving the submission when clicking on "Back."
-  Better error handling.
-  Prevent form from navigating to the previous or next step when updating the submission fails.

### Screenshots

Screenshot 2024-06-14 at 16 49 04

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-1072]: https://artsyproduct.atlassian.net/browse/ONYX-1072?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Sell/Components/BottomFormNavigation.tsx [hunk 1]
```diff
@@ -1,28 +1,54 @@
...
-    actions.goToNextStep()
+  const onBack = async () => {
+    setIsLoadingBack(true)
+
+    try {
+      await submitForm()
+
+      actions.goToPreviousStep()
+    } catch (error) {
+      console.error("Error submitting form", error)
```
Comment (olerichter00) [inline src/Apps/Sell/Components/BottomFormNavigation.tsx:26]: Todo: Use `logger` instead:

```
const logger = createLogger("BottomFormNavigation.tsx")
...
logger.error("Error submitting form, error) 
```


https://artsy.slack.com/archives/C05EQL4R5N0/p1718379105202439

Comment (olerichter00) [reply src/Apps/Sell/Components/BottomFormNavigation.tsx:26]: We should probably update the logger everywhere in the new submission flow.

File: src/Apps/Sell/Components/BottomFormNavigation.tsx [hunk 2]
```diff
@@ -1,63 +1,75 @@
...
+  return (
+    <Flex
+      width="100%"
+      p={[2, 4]}
+      flexDirection="row"
+      justifyContent="space-between"
+      alignItems="center"
+    >
+      <BottomFormBackButton />
+      <BottomFormNextButton />
```
Comment (olerichter00) [inline src/Apps/Sell/Components/BottomFormNavigation.tsx:15]: I'm separating both buttons to be able to render the loading spinner independently for each button when `isSubmitting` is true.

**Review by dblandin — APPROVED** [review]
Nice! This is looking great!

**Pull request merged**